### PR TITLE
feat(rum-core): add apmRequest config typescript type

### PR DESCRIPTION
**Pull request by ReneWerner87**
@hmdhk you had forgotten the typescript files, can you please check if it fits like this

the feature is great and we use it, thanks for your effort

### CODE REVIEW

**Comment by devcorpio** [comment]
Hi @ReneWerner87,

Thanks for your contribution!

**TL;DR:**

The types have to be a bit different:

```
    apmRequest?: (
      requestParams: {
        xhr: XMLHttpRequest
        url: string
        method: string
        payload: string | undefined
        headers: Record<string, string> | undefined
      },
    ) => boolean
```

**Context**:

When `apmRequest` is set the agent execute it on two different situations:

1.  when fetching central config. This only happens when [centralConfig](https://www.elastic.co/guide/en/apm/agent/rum-js/5.x/configuration.html#central-config) is set to true.
2.  when sending events/beacons to the APM server

On the case of the **central config** the arguments the agent provides are `xhr`, `url`, and `method`. (payload and headers are undefined).

On the case of the events/beacons the `payload` and `headers` arguments are provided, but `payload` type is a [string](https://github.com/elastic/apm-agent-rum-js/blob/main/packages/rum-core/src/common/apm-server.js#L338) and headers is [an object literal](https://github.com/elastic/apm-agent-rum-js/blob/main/packages/rum-core/src/common/apm-server.js#L92).

**Comment by ReneWerner87** [comment]
> Hi @ReneWerner87,
> 
> Thanks for your contribution!
> 
> **TL;DR:**
> 
> The types have to be a bit different:
> 
> ```
>     apmRequest?: (
>       requestParams: {
>         xhr: XMLHttpRequest
>         url: string
>         method: string
>         payload: string | undefined
>         headers: Record<string, string> | undefined
>       },
>     ) => boolean
> ```
> 
> **Context**:
> 
> When `apmRequest` is set the agent execute it on two different situations:
> 
> 1. when fetching central config. This only happens when [centralConfig](https://www.elastic.co/guide/en/apm/agent/rum-js/5.x/configuration.html#central-config) is set to true.
> 2. when sending events/beacons to the APM server
> 
> On the case of the **central config** the arguments the agent provides are `xhr`, `url`, and `method`. (payload and headers are undefined).
> 
> On the case of the events/beacons the `payload` and `headers` arguments are provided, but `payload` type is a [string](https://github.com/elastic/apm-agent-rum-js/blob/main/packages/rum-core/src/common/apm-server.js#L338) and headers is [an object literal](https://github.com/elastic/apm-agent-rum-js/blob/main/packages/rum-core/src/common/apm-server.js#L92).

thx, i changed the types

**Comment by devcorpio** [comment]
run elasticsearch-ci/docs

**Review by devcorpio — APPROVED** [review]
LGTM! 🎉 , This will be part of the next release we do.
